runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -323,8 +323,172 @@ J9::ARM64::TreeEvaluator::DIVCHKEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+      if (!node->isArrayChkReferenceArray1())
+         {
+
+         // Loading the Class Pointer -> classDepthAndFlags
+#ifdef OMR_GC_COMPRESSED_POINTERS
+         generateTrg1MemInstruction(cg, TR::InstOpCode::ldrw, node, tmp1Reg,
+               new (cg->trHeapMemory()) TR::MemoryReference(obj1Reg, (int32_t) TR::Compiler->om.offsetOfObjectVftField(), cg));
+#else
+         generateTrg1MemInstruction(cg, isLong ? TR::InstOpCode::ldrx : TR::InstOpCode::ldrw, node, tmp1Reg,
+               new (cg->trHeapMemory()) TR::MemoryReference(obj1Reg, TR::Compiler->om.offsetOfObjectVftField(), cg));
+#endif
+         TR::TreeEvaluator::generateVFTMaskInstruction(cg, node, tmp1Reg);
+         generateTrg1MemInstruction(cg, TR::InstOpCode::ldrw, node, tmp1Reg,
+               new (cg->trHeapMemory()) TR::MemoryReference(tmp1Reg, offsetof(J9Class, classDepthAndFlags), cg));
+
+         // We need to perform a X & OBJECT_HEADER_SHAPE_MASK
+         generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::andimmw, node, tmp1Reg, tmp1Reg, OBJECT_HEADER_SHAPE_MASK << J9AccClassRAMShapeShift);
+         generateCompareImmInstruction(cg, node, tmp1Reg, OBJECT_HEADER_SHAPE_POINTERS << J9AccClassRAMShapeShift);
+
+         if (NULL == snippetLabel)
+            {
+            snippetLabel = generateLabelSymbol(cg);
+            gcPoint = generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, node, snippetLabel, TR::CC_NE);
+            snippet = new (cg->trHeapMemory()) TR::ARM64HelperCallSnippet(cg, node, snippetLabel, node->getSymbolReference(), doneLabel);
+            cg->addSnippet(snippet);
+            }
+         else
+            generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, node, snippetLabel, TR::CC_NE);
+         }
+
+      // Object2 must be of reference component type array, otherwise throw exception
+      if (!node->isArrayChkReferenceArray2())
+         {
+#ifdef OMR_GC_COMPRESSED_POINTERS
+         generateTrg1MemInstruction(cg, TR::InstOpCode::ldrw, node, tmp1Reg,
+               new (cg->trHeapMemory()) TR::MemoryReference(obj2Reg, (int32_t) TR::Compiler->om.offsetOfObjectVftField(), cg));
+#else
+         generateTrg1MemInstruction(cg, isLong ? TR::InstOpCode::ldrx : TR::InstOpCode::ldrw, node, tmp1Reg,
+               new (cg->trHeapMemory()) TR::MemoryReference(obj2Reg, TR::Compiler->om.offsetOfObjectVftField(), cg));
+#endif
+         TR::TreeEvaluator::generateVFTMaskInstruction(cg, node, tmp1Reg);
+         generateTrg1MemInstruction(cg, TR::InstOpCode::ldrw, node, tmp1Reg,
+               new (cg->trHeapMemory()) TR::MemoryReference(tmp1Reg, offsetof(J9Class, classDepthAndFlags), cg));
+
+         loadConstant32(cg, node, J9AccClassRAMArray, tmp2Reg);
+         generateTrg1Src2Instruction(cg, TR::InstOpCode::andw, node, tmp2Reg, tmp1Reg, tmp2Reg);
+
+         TR::Instruction * i = generateCompareImmInstruction(cg, node, tmp2Reg, NULLVALUE);
```

Comment:
The variable `i` is not used afterwards.
You can use `generateCompareBranchInstruction(TR::InstOpCode::cbzx)` for comparing with zero to branch.